src/main/java/core/mate/academy/service/ExcavatorProducerImpl.java

```diff
@@ -0,0 +1,12 @@
+package core.mate.academy.service;
+
+import core.mate.academy.model.Excavator;
+import java.util.List;
+
+public class ExcavatorProducerImpl implements MachineProducer<Excavator> {
```

Comment:
The class ExcavatorProducerImpl correctly parameterizes MachineProducer with the Excavator type, ensuring that only Excavator or its subtypes can be produced.